nuxeo-java-client-test/src/test/java/org/nuxeo/client/ITDirectory.java

```diff
@@ -100,4 +102,45 @@ public void itCanCreateUpdateFetchDeleteDirectory() {
         directory.deleteEntry(result.getId());
     }
 
+    @Test
+    public void itCanSetOrderingPropertyInteger() {
+        Directory directory = nuxeoClient.directoryManager().directory("continent");
+
+        DirectoryEntry entryWithInteger = new DirectoryEntry();
+        entryWithInteger.setDirectoryName("continent");
+        entryWithInteger.putIdProperty("test1");
+        entryWithInteger.putLabelProperty("test1");
+        entryWithInteger.putObsoleteProperty(0);
+
+        entryWithInteger.putOrderingProperty(Integer.valueOf(Integer.MAX_VALUE).longValue());
+        DirectoryEntry resultWithInteger = directory.createEntry(entryWithInteger);
+        assertNotNull(resultWithInteger);
+
+        resultWithInteger = resultWithInteger.update();
+        assertEquals(Long.valueOf(Integer.MAX_VALUE), resultWithInteger.getOrderingProperty());
+
+        resultWithInteger.putOrderingProperty(Long.valueOf(Integer.MAX_VALUE));
+        assertEquals(Long.valueOf(Integer.MAX_VALUE), resultWithInteger.getOrderingProperty());
```

Comment:
Here what it is also interesting to test is the returned object of `putOrderingProperty`, also we can assert that between creation and update. For instance:
```suggestion
        entryWithInteger.putOrderingProperty(0L);
        DirectoryEntry resultWithInteger = directory.createEntry(entryWithInteger);
        assertNotNull(resultWithInteger);
        assertEquals(0L, resultWithInteger.getOrderingProperty());
        // assert the previous value and set a new Integer one
        assertEquals(0L, resultWithInteger.putOrderingProperty(Long.valueOf(Integer.MAX_VALUE)));

        resultWithInteger = resultWithInteger.update();
        assertEquals(Long.valueOf(Integer.MAX_VALUE), resultWithInteger.getOrderingProperty());
        // assert the previous value
        assertEquals(Long.valueOf(Integer.MAX_VALUE), resultWithInteger.putOrderingProperty(0L));
        
        resultWithInteger = resultWithInteger.update();
        assertEquals(0L, resultWithInteger.getOrderingProperty());
```